lib/ui/settings/data/import/analyze_qr_code.dart

```diff
@@ -0,0 +1,268 @@
+import 'package:ente_auth/utils/toast_util.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+import 'package:wechat_assets_picker/wechat_assets_picker.dart';
+
+class QrScanner extends StatefulWidget {
+  const QrScanner({super.key});
+
+  @override
+  State<QrScanner> createState() => _QrScannerState();
+}
+
+class _QrScannerState extends State<QrScanner> {
+  bool isNavigationPerformed = false;
+  bool isScannedByImage = false;
+
+  //Scanner Initialization
+  MobileScannerController scannerController = MobileScannerController(
+    detectionSpeed: DetectionSpeed.normal,
+    facing: CameraFacing.back,
+  );
+
+  @override
+  void dispose() {
+    scannerController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final l10n = context.l10n;
+    return SafeArea(
+      child: Scaffold(
+        backgroundColor: Colors.black,
+        body: Stack(
+          alignment: Alignment.topLeft,
+          children: [
+            Stack(
+              children: [
+                MobileScanner(
+                  controller: scannerController,
+                  onDetect: (capture) async {
+                    if (!isNavigationPerformed) {
+                      isNavigationPerformed = true;
+                      if (capture.barcodes[0].rawValue!
+                          .startsWith(kGoogleAuthExportPrefix)) {
+                        if (isScannedByImage) {
+                          final result = await showDialogWidget(
+                            context: context,
+                            title: l10n.reminderText,
+                            body: l10n.reminderPopupBody,
+                            buttons: [
+                              ButtonWidget(
+                                buttonType: ButtonType.primary,
+                                labelText: l10n.ok,
+                                isInAlert: true,
+                                buttonSize: ButtonSize.large,
+                                buttonAction: ButtonAction.first,
+                              ),
+                            ],
+                          );
+                          if (result?.action != null &&
+                              result!.action == ButtonAction.first) {
+                            isScannedByImage = false;
+                          }
+                        }
+                        HapticFeedback.vibrate();
+                        try {
+                          List<Code> codes =
+                              parseGoogleAuth(capture.barcodes[0].rawValue!);
+                          scannerController.dispose();
+                          Navigator.of(context).pop(codes);
+                        } catch (e) {
+                          debugPrint('Error occurred while parsing: $e');
```

Comment:
Let's log the exception using logger and also show a toast or generic error dialog.. Otherwise user won't get any feedback.

Reply:
Should we need to display the error directly to the user?